src/main/java/pl/cleankod/exchange/core/gateway/AccountService.java

```diff
@@ -0,0 +1,14 @@
+package pl.cleankod.exchange.core.gateway;
+
+import pl.cleankod.exchange.core.domain.Account;
+import pl.cleankod.exchange.entrypoint.model.AccountNotFoundException;
+import pl.cleankod.exchange.entrypoint.model.AccountResponse;
+
+import java.util.Currency;
+
+public interface AccountService {
```

Comment:
Why did you introduce the `AccountService`?

Did you notice what role the `...UseCase` classes have within this application?

And why did you put the `AccountService` class in the `gateway` package?